sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/AnySupport.scala

```diff
@@ -389,8 +389,13 @@ class AnySupport(
         ScalaPbAny(BytesPrimitive.fullName, primitiveToBytes(BytesPrimitive, byteString))
 
       case other =>
-        throw SerializationException(
-          s"Don't know how to serialize object of type ${other.getClass}. Try passing a protobuf, using a primitive type, or using a type annotated with @Jsonable.")
+        try {
+          ScalaPbAny.fromJavaProto(JsonSupport.encodeJson(value))
```

Comment:
Is there a risk that this accidentally json encodes something a user didn't mean to, given that it is the same code path for any response/forwarded/side effect message from any component? 

Should we narrow that down with some additional parameter/flag for AnySupport when serializing and only flag that for entity state?

Reply:
Indeed, I think we should drive it somehow. 

Preferably not with an extra annotation (like the old `@Jsonable`), but let the registration passes something down the road. 

One thing I had in mind is to introduce a `Serializer` that is then registered by the provider. By default it would be protobuf, for POJOs we switch to Jackson. The user could choose something else as well. I'm think on Scala devs that would want something else, like play-json or circe. 

Anyway, this need to mature.